cmd/collector/app/zipkin/json.go

```diff
@@ -149,6 +141,15 @@ func spanToThrift(s zipkinSpan) (*zipkincore.Span, error) {
 	return tSpan, nil
 }
 
+// id can be padded with zeros. We let it fail later in case it's longer than 32
+func cutLongID(id string) string {
+	l := len(id)
+	if l > 16 && l <= 32 {
+		return id[:16]
```

Comment:
afaik zipkin does not support span IDs longer than 64bit, so what's the use case for truncating?

Reply:
This is not used for trace ids. 

There can be span id `1` padded with 16 zeros.

Reply:
> There can be span id 1 padded with 16 zeros.

do you mean for a total of 17 digits? Doesn't sound like it would be valid for zipkin.

Reply:
Try this:
```json
[
  {
    "traceId": "7d3277b087a7c05a",
    "name": "/",
    "id": "000000000000000001",
    "timestamp": 1508924339988597,
    "duration": 1781,
    "annotations": [],
    "binaryAnnotations": [
      {
        "key": "lc",
        "value": "nginx",
        "endpoint": {
          "ipv4": "",
          "port": 0,
          "serviceName": "nginx"
        }
      }
    ]
  }
]
```

zipkin consumes it without problems. The ID parsing function what we had here before was exact impl from zipkin and there were tests for these types of ids.

Reply:
I still think it's invalid ID - what if it was padded with 1s instead of 0s? By truncating it you are essentially reading a different number.

If you only want to truncate the 0s, you may want to (a) optionally check that the chopped off characters are all 0s, and (b) parse the right-most 16 digits `id[l-16:]`.

Reply:
It certainly works in zipkin. Here is the zipkin code https://github.com/openzipkin/zipkin/blob/eeeeb3c02b0e0c578eb36c08d80cf2bf400067fc/zipkin/src/main/java/zipkin/internal/Util.java#L101

Reply:
you still need (b), zipkin does it too: `int beginIndex = length > 16 ? length - 16 : 0;`

Reply:
I did this
```
func cutLongID(id string) string {
 +	l := len(id)
 +	if l > 16 && l <= 32 {
 +		start := l - 16
 +		return id[start:]
 +	}
 +	return id
 +}
```